### PR TITLE
MAINT: linting statespace tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
@ChadFulton doing a few files instead of one at a time.

### CODE REVIEW

File: statsmodels/tsa/statespace/tests/test_tools.py [hunk 1]
```diff
@@ -395,11 +396,12 @@ def test_cases(self):
...
                 cov = np.eye(unconstrained.shape[0])
-            constrained, _ = tools.constrain_stationary_multivariate(unconstrained, cov)
+            constrained, _ = tools.constrain_stationary_multivariate(unconstrained, cov)  # noqa:E501
             companion = tools.companion_matrix(
-                [1] + [-np.squeeze(constrained[i]) for i in range(len(constrained))]
+                [1] + [-np.squeeze(constrained[i])
+                       for i in range(len(constrained))]
             ).T
-            assert_equal(np.max(np.abs(np.linalg.eigvals(companion))) < 1, True)
+            assert np.max(np.abs(np.linalg.eigvals(companion))) < 1
```
Comment (ChadFulton) [inline statsmodels/tsa/statespace/tests/test_tools.py]: I have not been keeping up with the discussion on this, but I thought we didn't like plain assert statements?

**Comment by ChadFulton** [comment]
>  I can't tell why the linter is now complaining about copulas.py

Oh I see, this is what #5080 is about?

**Comment by jbrockmendel** [comment]
> Oh I see, this is what #5080 is about?

Precisely.

File: statsmodels/tsa/statespace/tests/test_tools.py (hunk 1 above)
Comment (jbrockmendel) [reply statsmodels/tsa/statespace/tests/test_tools.py]: That's a relic from a long time ago, when running `python -A` (I think -A, not totally sure) would strip out assertions.  Now that we're using pytest, it rewrites assertions, so this more readable version is the way to go.

**Comment by jbrockmendel** [comment]
@ChadFulton next up on the approved-earlier train

File: statsmodels/tsa/statespace/tests/test_tools.py (hunk 1 above)
Comment (ChadFulton) [reply statsmodels/tsa/statespace/tests/test_tools.py]: @josef-pkt is this true? are `assert`s okay now?

**Comment by josef-pkt** [comment]
I still don't like the pytest asserts and prefer numpy's `assert_` (because I worry about pytest magic)
However, I let several plain assert go through in recent PRs. One reason is that I don't want to argue each time, the other reason is that keeping or adding a few cases works as experiment to see how robust they are.

For inequality testing I usually use numpy assert_array_less.

**Comment by jbrockmendel** [comment]
Is the startup time related to the assert rewriting?  I thought it was mainly about the test collection.

@ChadFulton the Travis fail is fixed by #5080, so I think we’re good to go.

**Comment by ChadFulton** [comment]
Still need to decide about the `assert`s in general. Do you mind reversing that part of the change so I can just merge this?

File: statsmodels/tsa/statespace/tests/test_tools.py (hunk 1 above)
Comment (jbrockmendel) [reply statsmodels/tsa/statespace/tests/test_tools.py]: This one I'll consent to change to an assert_array_less.  I think that's the only one here.

**Comment by ChadFulton** [comment]
Thanks! Will merge after green.

**Comment by jbrockmendel** [comment]
de-facto green

**Comment by ChadFulton** [comment]
Thanks @jbrockmendel

**Pull request merged**